lib/chef/http/basic_client.rb

```diff
@@ -139,7 +139,8 @@ def env
 
       def http_client_builder
         http_proxy = proxy_uri
-        if http_proxy.nil?
+        if !http_proxy.respond_to?(:host) || !http_proxy.respond_to?(:port) ||
+           !http_proxy.respond_to?(:user) || !http_proxy.respond_to?(:password)
```

Comment:
actually it should return nil, and as far as i can see the old code should pass the validation and it looks like some code might have gotten snagged from another project and stuffed into `#proxy_uri` which already fixed this use case?